src/posix/platform/netif.cpp

```diff
@@ -334,7 +334,7 @@ static void UpdateUnicastLinux(const otIp6AddressInfo &aAddressInfo, bool aIsAdd
 
     req.nh.nlmsg_len = NLMSG_ALIGN(req.nh.nlmsg_len) + rta->rta_len;
 
-    if (aAddressInfo.mIsAnycast)
+    if (!aAddressInfo.mPreferred)
```

Comment:
I cannot directly comment that line, but we need to set `ifa_cacheinfo.ifa_prefered` to zero to mark it not preferred. (we are setting only `ifa_valid`).

Reply:
And, update for `BSD` platforms as well?

Reply:
> I cannot directly comment that line, but we need to set `ifa_cacheinfo.ifa_prefered` to zero to mark it not preferred. (we are setting only `ifa_valid`).

It's  memset to `0`.

Reply:
> > I cannot directly comment that line, but we need to set `ifa_cacheinfo.ifa_prefered` to zero to mark it not preferred. (we are setting only `ifa_valid`).
> 
> It's memset to `0`.

1. It is not readable/maintainable to rely on default zero-initialization.
2. We need to set `ifa_cacheinfo.ifa_prefered` to non-zero value (e.g. `UINT32_MAX`) when the address is preferred.
3. Besides, I think `ifa_cacheinfo. ifa_valid ` should always be set to `UINT32_MAX` no matter this address is preferred or not (as it is always valid address).

Reply:
> > > I cannot directly comment that line, but we need to set `ifa_cacheinfo.ifa_prefered` to zero to mark it not preferred. (we are setting only `ifa_valid`).
> > 
> > 
> > It's memset to `0`.
> 
> 1. It is not readable/maintainable to rely on default zero-initialization.

It's not default zero-initialization, but explicitly memset to 0. If this pattern is not readable/maintainable, I guess we need to change a massive amount of code in OpenThread. I don't think it necessary to double set a field. Thoughts?

> 2. We need to set `ifa_cacheinfo.ifa_prefered` to non-zero value (e.g. `UINT32_MAX`) when the address is preferred.
> 3. Besides, I think `ifa_cacheinfo. ifa_valid ` should always be set to `UINT32_MAX` no matter this address is preferred or not (as it is always valid address).

Please note this attribute is only attached when this address is not preferred.